components/scream/src/physics/spa/spa_functions_impl.hpp

```diff
@@ -0,0 +1,294 @@
+
+  // Set up temporary arrays that will be used for the spa interpolation.
+  view_2d<Spack> p_src("p_mid_src",ncols_atm,pressure_state.nlevs), 
+                 ccn3_src("ccn3_src",ncols_atm,pressure_state.nlevs);
+  view_3d<Spack> aer_g_sw_src("aer_g_sw_src",ncols_atm,nswbands,pressure_state.nlevs),
+                 aer_ssa_sw_src("aer_ssa_sw_src",ncols_atm,nswbands,pressure_state.nlevs),
+                 aer_tau_sw_src("aer_tau_sw_src",ncols_atm,nswbands,pressure_state.nlevs),
+                 aer_tau_lw_src("aer_tau_lw_src",ncols_atm,nlwbands,pressure_state.nlevs);
+
+  using ExeSpace = typename KT::ExeSpace;
+  const Int nk_pack = ekat::npack<Spack>(nlevs_atm);
+  const auto policy = ekat::ExeSpaceUtils<ExeSpace>::get_default_team_policy(ncols_atm, nk_pack);
+  // SPA Main loop
+  // Parallel loop order:
+  // 1. Loop over all horizontal columns (i index)
+  // 2. Loop over all aerosol bands (n index) - where applicable
+  // 3. Loop over all vertical packs (k index)
+  Kokkos::parallel_for(
+    "spa main loop",
+    policy,
+    KOKKOS_LAMBDA(const MemberType& team) {
+
+    const Int i = team.league_rank();  // SCREAM column index
+
+    // Get single-column subviews of all 2D inputs, i.e. those that don't have aerosol bands
+    const auto& ps_beg_sub                 = pressure_state.ps_this_month(i);
+    const auto& ps_end_sub                 = pressure_state.ps_next_month(i);
+    const auto& pmid_sub                   = ekat::subview(pressure_state.pmid, i);
+    const auto& p_src_sub                  = ekat::subview(p_src, i);
+
+    const auto& ccn3_beg_sub               = ekat::subview(data_beg.CCN3, i);
+    const auto& ccn3_end_sub               = ekat::subview(data_end.CCN3, i);
+    const auto& ccn3_src_sub               = ekat::subview(ccn3_src, i);
+
+    // First Step: Horizontal Interpolation if needed - Skip for Now
+  
+    // Second Step: Temporal Interpolation
+    // Use basic linear interpolation function y = b + mx
+    auto t_norm = (t_now-t_beg)/t_len;
+    /* Determine PS for the source data at this time */
+    auto ps_src = linear_interp(ps_beg_sub,ps_end_sub,t_norm);
+    {
+    /* Reconstruct the vertical pressure profile for the data and time interpolation
+     * of the data.
+     * Note: CCN3 has the same dimensions as pressure so we handle that time interpolation
+     *       in this loop as well.  */
+    using C = scream::physics::Constants<Real>;
+    static constexpr auto P0 = C::P0;
+    const Int nk_pack = ekat::npack<Spack>(pressure_state.nlevs);
+    Kokkos::parallel_for(
+      Kokkos::TeamThreadRange(team, nk_pack), [&] (Int k) {
+        // Reconstruct vertical temperature profile using the hybrid coordinate system.
+        p_src_sub(k)    = ps_src * pressure_state.hybm(k) + P0 * pressure_state.hyam(k);
+        // Time interpolation for CCN3
+        ccn3_src_sub(k) = linear_interp(ccn3_beg_sub(k),ccn3_end_sub(k),t_norm);
+    });
+    team.team_barrier();
+    }
```

Comment:
What's the point of these scope blocks? I don't think they are needed, for they don't help with temps scoping. Besides, the indentation is off. Can we remove them?